pac4j-http/src/main/java/org/pac4j/http/credentials/TokenCredentials.java

```diff
@@ -42,4 +42,11 @@ public String getToken() {
     public String toString() {
         return CommonHelper.toString(this.getClass(), "token", this.token, "clientName", getClientName());
     }
+
+    @Override
+    public void clear() {
+        this.token = null;
+        this.setClientName(null);
+        this.setUserProfile(null);
```

Comment:
If ever we think we need to clear the user profile, you should call the `clear` method of the user profile.